automation/setup-local-env-for-e2e.sh

```diff
@@ -0,0 +1,425 @@
+
+AUTO_USERS_PASSWD=$(docker run --rm -e VAULT_TOKEN=$VAULT_TOKEN             \
+    ${DSDE_TOOLBOX_DOCKER_IMAGE}                                            \
+    vault read --format=json ${FC_USERS_PATH}                               \
+    | jq -r .data.automation_users_passwd)
+
+echo "export AUTO_USERS_PASSWD=${AUTO_USERS_PASSWD}" >> ${TEST_RESOURCES}/$e2eEnv
+
+USERS_PASSWD=$(docker run --rm -e VAULT_TOKEN=$VAULT_TOKEN                  \
+    ${DSDE_TOOLBOX_DOCKER_IMAGE}                                            \
+    vault read --format=json ${FC_USERS_PATH}                               \
+    | jq -r .data.users_passwd)
+
+echo "export USERS_PASSWD=${USERS_PASSWD}" >> ${TEST_RESOURCES}/$e2eEnv
+
+# Function to obtain user tokens
+obtainUserTokens() {
+    echo "Obtaining user tokens for hermione, harry, and ron..."
+    hermione=$(docker run --rm -v ${SCRIPT_DIR}/src:/src             \
+            -i gcr.io/oauth2l/oauth2l fetch                          \
+            --credentials /src/test/resources/firecloud-account.json \
+            --scope profile,email,openid                             \
+            --email hermione.owner@quality.firecloud.org)
+
+    harry=$(docker run --rm -v ${SCRIPT_DIR}/src:/src                \
+            -i gcr.io/oauth2l/oauth2l fetch                          \
+            --credentials /src/test/resources/firecloud-account.json \
+            --scope profile,email,openid                             \
+            --email harry.potter@quality.firecloud.org)
+
+    ron=$(docker run --rm -v ${SCRIPT_DIR}/src:/src                  \
+            -i gcr.io/oauth2l/oauth2l fetch                          \
+            --credentials /src/test/resources/firecloud-account.json \
+            --scope profile,email,openid                             \
+            --email ron.weasley@quality.firecloud.org)
+
+    USERS_METADATA_JSON="[
+      {
+        \"email\":\"hermione.owner@quality.firecloud.org\",
+        \"type\":\"owner\",
+        \"bearer\":\"${hermione}\"
+      },
+      {
+        \"email\":\"harry.potter@quality.firecloud.org\",
+        \"type\":\"student\",
+        \"bearer\":\"${harry}\"
+      },
+      {
+        \"email\":\"ron.weasly@quality.firecloud.org\",
+        \"type\":\"student\",
+        \"bearer\":\"${ron}\"
+      }
+    ]"
+
+    # Try the -w0 option (without line breaks) first
+    if base64 --help | grep -q '\-w'; then
+      USERS_METADATA_JSON_B64=$(printf '%s' "${USERS_METADATA_JSON}" | base64 -w0)
+    # If -w0 is not available, try the -b0 option (also without line breaks)
+    elif base64 --help | grep -q '\-b'; then
+      USERS_METADATA_JSON_B64=$(printf '%s' "${USERS_METADATA_JSON}" | base64 -b 0)
+    else
+      echo "Error: No suitable base64 encoding option found."
+      exit 1
+    fi
+
+    echo "export USERS_METADATA_B64=\"${USERS_METADATA_JSON_B64}\"" >> ${TEST_RESOURCES}/$e2eEnv
+}
+
+attachLandingZoneToBillingProject() {
```

Comment:
I worry about duplicating all of this… isn't it basically `attach-landing-zone-to-bee.yaml` github workflow? One can't just run the workflow against the bee that they wish to use for running the tests locally?

Reply:
I don't think that would work: isn't this setting environmental variables so a locally running instance of rawls can connect to a remote bee?

Reply:
No, I'm talking about the code starting at line 338 that is hitting orch (running in a bee) to create a billing project, and then polling on the billing project creation status.

Reply:
You can provide the billing project option `--billingProject <value>` if you use the `attach-billing-project-to-landing-zone.yaml` workflow in `terra-github-workflows` to attach it to the LZ. The method `attachBillingProjectToLandingZone` will be bypassed if the billing project is provided.

Reply:
So do we think there is sufficient value in duplicating all this code related to creating a billing project and polling for the billing project to complete? Again, I worry that this pattern is going to result in the copies of this code in every repository that wishes to use the `attach-landing-zone-to-billing-project` action.

Reply:
I agree with you on this point. We should make `--billingProject` a required argument and it will up to the developer to provide a valid billing project already attached to a landing zone.

Reply:
Simplified this setup script. Make `billingProject` required. We no longer need Azure Coordinates setup in this script.